### PR TITLE
10-7959c OHI Email Required

### PR DESCRIPTION
**Pull request by reiting**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

If the folder you changed contains a `manifest.json`, search for its `entryName` in the content-build [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) (the `entryName` there will match).

If an entry for this folder exists in content-build and you are:
1. **Deleting a folder**:
   1. First search `vets-website` for _all_ instances of the `entryName` in your `manifest.json` and remove them in a separate PR. Look particularly for references in `src/applications/static-pages/static-pages-entry.js` and `src/platform/forms/constants.js`. _**If you do not do this, other applications will break!**_
      - _Add the link to your merged vets-website PR here_
   2. Then, Delete the application entry in [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) and merge that PR **before** this one
      - _Add the link to your merged content-build PR here_

2. **Renaming or moving a folder**: Update the entry in the [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json), but do not merge it until your vets-website changes here are merged. The content-build PR must be merged immediately after your vets-website change is merged in to avoid CI errors with content-build (and Tugboat).

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary
This PR adds an email field to the OHI form that is required.

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/96997

## Testing done
unit
e2e

## Screenshots
Screenshot 2024-12-11 at 2 14 28 PM

## What areas of the site does it impact?
this form only

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback
NA

### CODE REVIEW

File: src/applications/ivc-champva/10-7959C/chapters/formSignature.js [hunk 1]
```diff
@@ -9,6 +11,21 @@ export const formSignatureSchema = {
...
+            }`,
+          };
+
+          return {
+            'ui:title': `Which of these best describes you?`,
+            'ui:options': {
+              labels,
+            },
+          };
+        },
```
Comment (michaelclement) [inline src/applications/ivc-champva/10-7959C/chapters/formSignature.js:28]: I believe this function should look a little different as of [this recent PR](https://github.com/department-of-veterans-affairs/vets-website/pull/33388/files#diff-b07cb510c6203f7cec2d4a2a6a07960dc5079f7e9c6c9875eec81558cf94cc77) - we should be able to get rid of the `updateUiSchema` function here and just have the label keys

File: src/applications/ivc-champva/10-7959C/config/form.js [hunk 2]
```diff
@@ -128,25 +131,35 @@ const formConfig = {
     formSignature: {
       title: 'Signer information',
       pages: {
-        formSignature: {
-          path: 'form-signature',
+          formSignature: {
+          path: UPLOADS_COMPLETE_PATH,
+          // initialData: mockdata.data,
```
Comment (michaelclement) [inline src/applications/ivc-champva/10-7959C/config/form.js:136]: Same issue here, I `UPLOADS_COMPLETE_PATH` should be the string added in [this previous PR](https://github.com/department-of-veterans-affairs/vets-website/pull/33388/files#diff-07448f852c91e9cc2cee14e9104a04bc51be6afa5da16c572969be0c1c86c280)

File: src/applications/ivc-champva/10-7959C/config/form.js [hunk 3]
```diff
@@ -128,25 +131,35 @@ const formConfig = {
...
             `${
               formData.certifierRole === 'applicant' ? 'Your' : 'Beneficiary’s'
             } name`,
           ...applicantNameDobSchema,
         },
+      },
+    },
+    applicantInformation: {
+      title: 'Beneficiary information',
+      pages: {
```
Comment (michaelclement) [inline src/applications/ivc-champva/10-7959C/config/form.js:162]: This looks like we may have two duplicate chapter headers - could be result from a merge conflict?

File: src/applications/ivc-champva/10-7959C/chapters/formSignature.js [hunk 4]
```diff
@@ -27,3 +44,20 @@ export const formSignatureSchema = {
...
+    applicantEmail: emailUI(),
+  },
+  schema: {
+    type: 'object',
+    required: ['applicantEmail'],
+    properties: {
+      applicantEmail: emailSchema,
+    },
+  },
+};
```
Comment (michaelclement) [inline src/applications/ivc-champva/10-7959C/chapters/formSignature.js:63]: With the inclusion of email we'll want to add the email property into the `primaryContactInfo` object [in the submit transformer](https://github.com/department-of-veterans-affairs/vets-website/blob/4cb5b2c08493835088a44e98684c021c0437eb07/src/applications/ivc-champva/10-7959C/config/submitTransformer.js#L11) as well.

File: src/applications/ivc-champva/10-7959C/config/submitTransformer.js [hunk 5]
```diff
@@ -8,7 +8,7 @@ function getPrimaryContact(data) {
   // treated as the primary contact.
   return {
     name: data?.applicantName ?? false,
-    email: false, // We don't collect email
+    email: data?.applicantEmail,
```
Comment (michaelclement) [inline src/applications/ivc-champva/10-7959C/config/submitTransformer.js:11]: This unit test is failing - I think this line just needs to be `data?.applicantEmail ?? false,` and that should fix it. Thanks!

Screenshot 2024-12-13 at 08 03 42

**Review by michaelclement — DISMISSED** [review]

File: src/applications/ivc-champva/10-7959C/config/form.js [hunk 6]
```diff
@@ -129,17 +132,22 @@ const formConfig = {
...
       pages: {
         formSignature: {
+          // initialData: mockdata.data,
           path: 'form-signature',
           title: 'Form signature',
           ...formSignatureSchema,
         },
+        signerEmail: {
+          path: 'signer-email',
+          title: 'Your email Address',
```
Comment (jamigibbs) [inline src/applications/ivc-champva/10-7959C/config/form.js:142]: Should the "A" be lower case here?

**Review by jamigibbs — DISMISSED** [review]

File: src/applications/ivc-champva/10-7959C/config/form.js (hunk 6 above)
Comment (reiting) [reply src/applications/ivc-champva/10-7959C/config/form.js]: Yes....yes it should. Somehow several of us missed this one. So sorry, that's my bad! I've fixed it. Let me know if you need me to get another team review for it. :)

**Review by jamigibbs — APPROVED** [review]

**Pull request merged**